pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/ddp/kitrequest/KitRequestDao.java

```diff
@@ -134,6 +139,31 @@ public String getKitLabelFromDsmKitRequestId(long dsmKitRequestId) {
         return (String) results.resultValue;
     }
 
+    public String getKitTypeByKitRequestId(long kitRequestId) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_KIT_TYPE_BY_KIT_REQUEST_ID)) {
+                stmt.setLong(1, kitRequestId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(DBConstants.KIT_TYPE_NAME);
+                    } else {
+                        dbVals.resultValue = null;
+                        dbVals.resultException = new DsmInternalError("No kit request found for kit_request_id " + kitRequestId);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Couldn't get kit type for kit request: " + kitRequestId, results.resultException);
```

Comment:
Pls use DsmInternalError or DSMBadRequestException or similar in place of RuntimeException, which is too general.

Also, this pattern can be simplified. You are expecting to return a value, otherwise it is an error. 

Here is an example. (It's not identical, but an illustration of what simpler looks like.)

```    
public static AdminOperationResponse.AdminOperationResult getOperationRecord(int operationId) {
        return inTransaction(conn -> {
            AdminOperationResponse.AdminOperationResult result = null;
            try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_OPERATION_BY_ID)) {
                stmt.setInt(1, operationId);
                try (ResultSet rs = stmt.executeQuery()) {
                    if (rs.next()) {
                        result = getResult(rs);
                    }
                }
            } catch (SQLException e) {
                throw new DsmInternalError("Error getting admin operation result for operation " + operationId, e);
            }
            return result;
        });
    }
```

Reply:
It looks like you chose to not simplify this method?

Reply:
overlooked and just edited the Exception. refactored it now.